c-deps/proj: bump PROJ to compile on more systems

otan opened the pull request:
CMake is broken for the value of `PTHREAD_MUTEX_RECURSIVE` on certain
operating systems. To rectify this, we've patched PROJ at version 4.9.3
at `cc4a178` with a fix. We'll be using the CRDB fork of PROJ for the
meantime to fix this for now.

fixes #49749

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49755)
<!-- Reviewable:end -->

knz reviewed (approved):
do you want to add "fixes #49749" to that pr desc?

otan commented:
done! tftr and for debugging the fix.

bors r=knz

otan commented:
bors r-

hmm, verify archive is not happy

otan commented:
`CMake Error at CMakeLists.txt:64 (CHECK_C_SOURCE_COMPILES):`

otan committed:
c-deps/proj: bump PROJ to compile on more systems

CMake is broken for the value of `PTHREAD_MUTEX_RECURSIVE` on certain
operating systems. To rectify this, we've patched PROJ at version 4.9.3
at `cc4a178` with a fix. We'll be using the CRDB fork of PROJ for the
meantime to fix this for now.

Release note: None

otan commented:
looks like include the module in CMakefiles.txt worked

bors r=knz

knz commented:
thanks oliver :100:
